<a href="https://colab.research.google.com/github/karolmorenole-cell/BachCDU3R2.ipynb/blob/main/BachCDU3R2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Bienvenido a este cuaderno de **R** en Google Colab para la U3 R2 de la Materia de Cultura Digital.

# Comenzaremos instalando el paquete **psych** y la libreria *psych* los cuales nos permitirán usar la función *describe*.


In [ ]:
#Instalando el paquete "psych"
install.packages("psych")

#Invocando a la librería "psych"
library(psych)

Ahora vamos a cargar un archivo al entorno de trabajo. Para ello nos desplazamos hacía los íconos del menú lateral izquierdo, hacemos clic izquierdo en el símbolo de la carpeta (*Archivos*) y damos clic en el ícono de *subir al almacenamiento de la sesión* (primer ícono del menú gráfico debajo del texto *Archivos*).

En la ventana emergente elegimos el archivo **Concentrado.xlsx**, y esperamos a que se cargue.

Para trabajar nuestro archivo en Excel requerimos instalar el paquete **readxl** e invocar la librería.

In [ ]:
install.packages("readxl")
library(readxl)

Para proceder con el análisis estadístico debemos asignar nuestro archivo a un *data frame* o variable.

Luego invocaremos las funciones *describe* y *summary* de la librería *psych*.

In [28]:
#Asignando nuestro archivo a la variable df
df <- read_excel("Copia de Concentrado.xlsx")

#Nota, en caso de que tu archivo tenga más de una hoja puedes especificar
#la hoja a leer con df <- read_excel("Concentrado.xlsx", sheet = x), reemplazando x
#por el número de hoja.

In [29]:
#Ejecutando la función describe() sobre df
describe(df)

,vars,n,mean,sd,median,trimmed,mad,min,max,range,skew,kurtosis,se
,<int>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
Registro,1,489,249.922290,144.7563459,252,249.905852,186.8076,1,499,498,-0.002130511,-1.21822807,6.54610824
Edad,2,446,24.558296,6.9256054,23,23.474860,2.9652,15,65,50,2.426063377,8.00654299,0.32793692
Género*,3,453,1.626932,0.4841548,2,1.658402,0.0000,1,2,1,-0.523185462,-1.73008153,0.02274757
Grupo*,4,489,3.382413,1.0157457,4,3.567430,0.0000,1,4,3,-1.237098184,-0.07124808,0.04593361
Extraversión,5,489,29.527607,4.2887284,30,29.753181,4.4478,15,39,24,-0.526340152,0.29088791,0.19394300
Cordialidad / Amabilidad,6,489,32.157464,4.4077231,32,32.305344,4.4478,16,45,29,-0.352757217,0.56952209,0.19932413
Responsabilidad (Tesón),7,489,33.529652,4.4730111,34,33.801527,4.4478,16,43,27,-0.729744857,1.09720088,0.20227655
Estabilidad Emocional,8,489,20.022495,3.9116138,20,19.847328,4.4478,11,35,24,0.478257069,0.16249373,0.17688929
Apertura a la Experiencia,9,489,41.388548,5.5780924,42,41.959288,4.4478,15,50,35,-1.710098970,5.11888692,0.25225006


In [30]:
#Ejecutando la función describe() sobre df
summary(df)

    Registro          Edad             Género          Grupo    
 Min.   :  1.0   Min.   :15.00   Length   :489   Length   :489  
 1st Qu.:124.0   1st Qu.:21.00   N.unique :  2   N.unique :  4  
 Median :252.0   Median :23.00   N.blank  :  0   N.blank  :  0  
 Mean   :249.9   Mean   :24.56   Min.nchar:  1   Min.nchar:  1  
 3rd Qu.:375.0   3rd Qu.:26.00   Max.nchar:  1   Max.nchar:  1  
 Max.   :499.0   Max.   :65.00   NAs      : 36                  
                 NAs    :43                                     
  Extraversión   Cordialidad / Amabilidad Responsabilidad (Tesón)
 Min.   :15.00   Min.   :16.00            Min.   :16.00          
 1st Qu.:27.00   1st Qu.:30.00            1st Qu.:31.00          
 Median :30.00   Median :32.00            Median :34.00          
 Mean   :29.53   Mean   :32.16            Mean   :33.53          
 3rd Qu.:33.00   3rd Qu.:35.00            3rd Qu.:37.00          
 Max.   :39.00   Max.   :45.00            Max.   :43.00          
                  

**NOTA**: Anque Colab suele guardar tus archivos en automático, es recomendable utilizar el comando *Ctrl + s* para guardar cambios, y aplicarlo antes de salir del archivo.

# Conclusión

En esta práctica utilizamos Google Colab y el lenguaje R para analizar los datos del archivo Concentrado.xlsx, el cual contiene las respuestas de un test de personalidad aplicado a un grupo de 489 alumnos, quienes tienen en promedio 24.5 años de edad. Al ejecutar los comandos summary() y describe(), logré obtener las estadísticas del grupo.
Esto fue lo más importante que encontré en los resultados:
 Lo más alto: El rasgo en el que el grupo obtuvo puntajes más altos fue Responsabilidad (Tesón) con un promedio de 33.52, seguido por Cordialidad / Amabilidad con 32.15. Esto significa que, en general, los alumnos se consideran responsables, organizados y empáticos.
 El punto medio: Los rasgos de Apertura a la Experiencia (promedio de 41.38) y Extraversión (promedio de 29.92) se mantuvieron en niveles intermedios y equilibrados.
 Lo más bajo: El promedio más bajo de todo el grupo fue el de Estabilidad Emocional, con apenas 20.02. Esto nos dice que una gran parte de los alumnos suele lidiar con niveles altos de estrés, sensibilidad o ansiedad.